### PR TITLE
Now can create annotations for symbollic alleles and masked alleles.

### LINKED ISSUE #5402: Funcotator crashes on symbolic alleles
`GencodeFuncotationFactory.createFuncotationsOnVariant()` (and possibly other places as well) needs to detect symbolic alleles and handle them appropriately. Should add a good test with, eg., the output of GCNV.

### PR DESCRIPTION
**Pull request by jonn-smith**
- Updated data sources to include variant sites for symbolic alleles.
- Fixed tests to be correct for new logic.
- Now has tests for symbollic alternate alleles and masked alleles.

Fixes #5402

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java [hunk 1]
```diff
@@ -694,9 +696,14 @@ private GencodeFuncotation createGencodeFuncotationOnTranscript(final VariantCon
...
+        if (altAllele.isSymbolic() || altAllele.equals(Allele.SPAN_DEL) ) {
+            return createFuncotationForSymbolicAltAllele(variant, altAllele, transcript, reference);
+        }
+
+        // If the reference allele or alternate allele contains any masked bases:
+        if ( variant.getReference().getBaseString().contains("N") || altAllele.getBaseString().contains("N") ) {
+            return createFuncotationForMaskedBases(variant, altAllele, transcript, reference);
         }
 
         // TODO: check for complex INDEL and warn and skip.
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java]: Can you file an issue for this TODO?

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java [hunk 2]
```diff
@@ -2121,33 +2133,107 @@ private GencodeFuncotation createIgrFuncotation(final VariantContext variant,
     }
 
     /**
-     * Creates a {@link GencodeFuncotation}s based on a given spanning deletion {@link Allele}.
+     * Creates a {@link GencodeFuncotation} based on a given symbolic alternate {@link Allele}.
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java]: Can we have one more sentence here? Such as "Creates a COULD NOT DETERMINE variant classification."

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java [hunk 3]
```diff
@@ -2121,33 +2133,107 @@ private GencodeFuncotation createIgrFuncotation(final VariantContext variant,
...
-    private GencodeFuncotation createFuncotationForSpanningDeletion(final VariantContext variant,
-                                                                    final String annotationTranscript,
-                                                                    final ReferenceContext reference){
+    private GencodeFuncotation createFuncotationForMaskedBases(final VariantContext variant,
+                                                             final Allele altAllele,
+                                                             final GencodeGtfTranscriptFeature annotationTranscript,
+                                                             final ReferenceContext reference) {
+
+        final StringBuilder alleleLogStringBuilder = new StringBuilder();
+        if ( variant.getReference().getBaseString().contains("N") ) {
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java]: Magic constant.  I bet there is already a constant defined for "N".  Even if not, it should not be inline here.

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java (hunk 1 above)
Comment (jonn-smith) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java]: Filed as #5411

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java (hunk 2 above)
Comment (jonn-smith) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java]: Sure.  Fixed!

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java (hunk 3 above)
Comment (jonn-smith) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java]: There doesn't seem to be one.  It should really live in `Allele.java`.  I'll add an issue to HTSJDK I guess.

In the meantime, I've created my own that will need to be removed upon completion of that issue.

Comment (jonn-smith) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java]: Added in HTSJDK as issue 1226 (https://github.com/samtools/htsjdk/issues/1226)

**Pull request merged**